src/ledger_state.rs

```diff
@@ -2074,32 +2084,145 @@ impl<'a, L: 'static + Ledger> LedgerState<'a, L> {
     }
 }
 
-pub struct LedgerStateStore<'a, L:Ledger+DeserializeOwned+Serialize> {
-    store: RollingLog::<EncryptingResourceAdapter<LedgerState<'a, L>>>
-}
+mod serde_ark_unchecked {
+    use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+    use serde::{
+        de::{Deserialize, Deserializer},
+        ser::{Serialize, Serializer},
+    };
+    use std::sync::Arc;
+
+    pub fn serialize<S: Serializer, T: CanonicalSerialize>(
+        t: &Arc<T>,
+        s: S,
+    ) -> Result<S::Ok, S::Error> {
+        let mut bytes = Vec::new();
+        t.serialize_unchecked(&mut bytes).unwrap();
+        Serialize::serialize(&bytes, s)
+    }
 
+    pub fn deserialize<'a, D: Deserializer<'a>, T: CanonicalDeserialize>(
+        d: D,
+    ) -> Result<Arc<T>, D::Error> {
+        let bytes = <Vec<u8> as Deserialize<'a>>::deserialize(d)?;
+        Ok(Arc::new(T::deserialize_unchecked(&*bytes).unwrap()))
+    }
+}
 
-impl<'a, L:Ledger+DeserializeOwned+Serialize> LedgerStateStore<'a,L> {
-    pub fn load(&self) -> Result<LedgerState<'a, L>, KeystoreError<L>> {
-       Ok( self.store.load_latest()?)
+/// Serialization intermediate for the static part of a ledger state.
+// // #[derive(Deserialize, Serialize, Debug)]
+#[derive(Deserialize, Serialize)]
+pub(crate) struct StaticState<'a> {
+    #[serde(with = "serde_ark_unchecked")]
+    proving_keys: Arc<ProverKeySet<'a, OrderByOutputs>>,
+}
 
+impl<'a, L: Ledger> From<&LedgerState<'a, L>> for StaticState<'a> {
+    fn from(w: &LedgerState<'a, L>) -> Self {
+        Self {
+            proving_keys: w.proving_keys.clone(),
+        }
     }
+}
+
+/// Serialization intermediate for the dynamic part of a ledger state.
+#[derive(Deserialize, Serialize)]
+#[serde(bound = "")]
+pub(crate) struct DynamicState<L: Ledger> {
+    now: EventIndex,
+    validator: Validator<L>,
+    record_mt: SparseMerkleTree,
+    nullifiers: NullifierSet<L>,
+}
 
-    pub fn update(&mut self, new_state: &LedgerState<'a, L>) -> Result<(),KeystoreError<L>> {
-       Ok(self.store.store_resource(new_state)?)
+impl<'a, L: Ledger> From<&LedgerState<'a, L>> for DynamicState<L> {
+    fn from(w: &LedgerState<'a, L>) -> Self {
+        Self {
+            now: w.now,
+            validator: w.validator.clone(),
+            record_mt: w.record_mt.clone(),
+            nullifiers: w.nullifiers.clone(),
+        }
     }
 }
 
-fn u256_to_signed(u: U256) -> BigInt {
-    let mut bytes = [0; 32];
-    u.to_little_endian(&mut bytes);
-    BigInt::from_bytes_le(Sign::Plus, &bytes)
+/// Storage for the ledger state.
+pub struct LedgerStateStore<'a, L: Ledger> {
```

Comment:
I think for consistency, this should be called `LedgerState`, since even though it doesn't follow the CRUD pattern, it's standing in for the `LedgerState` resource, and so it fits in the same category as `Assets`, `Accounts`, etc.

Reply:
I changed it to `LedgerStates` since there's also a `LedgerState`. The former includes two parts (the dynamic and the static stores), and the latter has all the state fields.